datadog/fwprovider/resource_datadog_software_catalog.go

```diff
@@ -254,6 +242,15 @@ func (r *catalogEntityResource) ImportState(ctx context.Context, request resourc
 	resource.ImportStatePassthroughID(ctx, frameworkPath.Root("id"), request, response)
 }
 
+func decodeBase64String(data string, response *resource.ReadResponse) []byte {
+	bytes, err := base64.StdEncoding.DecodeString(data)
+	if err != nil {
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error decoding base64 string"))
+		return nil
+	}
+	return bytes
+}
```

Comment:
Nit, Could we remove the response argument and just return the error from the function so the execution stops rather than continuing with nil in the caller